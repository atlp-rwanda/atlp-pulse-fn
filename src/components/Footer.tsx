import React, { useEffect, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import '../App';
import {
  FaFacebook,
  FaInstagram,
  FaTwitter,
  FaLinkedin,
  FaPlayCircle,
} from 'react-icons/fa';
import i18next from 'i18next';
import LogoFooter from '../assets/logo.svg';
import getLanguage from '../utils/getLanguage';
import LogoIcon from './logoIcon';

function Footer({ styles }: any) {
  const { t } = useTranslation();
  const lan = getLanguage();
  const userLang = window.navigator.language;
  const changeLan = (e: { target: { value: string } }) => {
    const { value } = e.target;
    i18next.changeLanguage(value);
  };
  const lanRef = useRef<any>();

  useEffect(() => {
    if (lanRef.current) {
      lanRef.current.value = lan;
    }
  }, []);
  return (
    <div className={`w-full bg-indigo-100 dark:bg-dark-bg ${styles} page-footer`}>
      <div className="px-2 flex flex-col lg:flex-row justify-between items-center w-full h-full">
        <div className="flex flex-col lg:flex-row items-center py-5">
          <div className="flex flex-col mr-12 lg:mr-0">
            <div className="flex mb-2 ml-10 text-dark dark:text-dark-text-fill">
              <LogoIcon />
              <h1
                className="text-3xl font-bold"
                data-testid="pulse"
              >
                PULSE
              </h1>
            </div>
            <div className="flex border-none justify-around items-center ml-16 mt-4 cursor-pointer">
              <FaFacebook className=" fa-xs " />
              <FaInstagram className=" " />
              <FaTwitter className="mr-1 " />
              <FaLinkedin className="mr-1 " />
              <FaPlayCircle className="" />
            </div>
          </div>
          <div className="flex flex-col py-4 lg:py-0 lg:flex-row">
            <div className="flex py-4">
              <ul className="lg:flex lg:flex-col mr-[8vh] md:mr-[16vh] lg:mr-0 lg:ml-28 cursor-pointer">
                <li className="py-2 text-sm font-bold">{t('Dev Pulse')}</li>
                <li className="py-2 text-xs">{t('About us')}</li>
                <li className="py-2 text-xs">{t('Contact us')}</li>
              </ul>
              <ul className="lg:flex lg:flex-col ml-[8vh] md:ml-[16vh] lg:ml-28 cursor-pointer">
                <li className="py-2 text-sm font-bold">{t('Product')}</li>
                <li className="py-2 text-xs">{t('Features')}</li>
                <li className="py-2 text-xs">{t('Integrations')}</li>
              </ul>
            </div>
            <div className="flex py-4">
              <ul className="lg:flex lg:flex-col mr-[8vh] md:mr-[16vh] lg:mr-0 lg:ml-28 cursor-pointer">
                <li className="py-2 text-sm font-bold">{t('Resources')}</li>
                <li className="py-2 text-xs">{t('Community')}</li>
                <li className="py-2 text-xs">{t('Help Center')}</li>
              </ul>
              <ul className="lg:flex lg:flex-col ml-[8vh] md:ml-[16vh] lg:ml-28 cursor-pointer">
                <li className="py-2 text-sm font-bold">
                  {t('Basic Settings')}
                </li>
                <li className="py-2 text-xs">{t('Terms and conditions')}</li>
                <li className="py-2 text-xs">{t('Privacy and Policies')}</li>
                <select
                  name="language"
                  defaultValue={userLang}
                  data-testid="lanChange"
                  ref={lanRef}
                  onChange={(e) => {
                    changeLan(e);
                  }}
                  className="mt-2 outline rounded-md px-2 py-1 dark:text-dark-text-fill dark:bg-dark-bg "
                >
                  <option value="en">English</option>
                  <option value="kn">Kinyarwanda</option>
                  <option value="fr">Français</option>
                </select>
              </ul>
            </div>
          </div>
        </div>
        <div className=" lg:flex mr-10">
          <span className="px-4 lg:py-3 cursor-pointer text-lg">
            © {new Date().getFullYear()} Pulse Technologies
          </span>
        </div>
      </div>
    </div>
  );
}

export default Footer;
