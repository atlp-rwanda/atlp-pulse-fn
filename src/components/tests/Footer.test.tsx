import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Footer from '../../components/Footer';

describe('<Footer/>', () => {
  it('Renders Header', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
