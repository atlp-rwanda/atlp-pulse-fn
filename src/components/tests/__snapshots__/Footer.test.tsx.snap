// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> Renders Header 1`] = `
<div
  className="w-full bg-indigo-100 dark:bg-dark-bg undefined page-footer"
>
  <div
    className="px-2 flex flex-col lg:flex-row justify-between items-center w-full h-full"
  >
    <div
      className="flex flex-col lg:flex-row items-center py-5"
    >
      <div
        className="flex flex-col mr-12 lg:mr-0"
      >
        <div
          className="flex mb-2 ml-10 text-dark dark:text-dark-text-fill"
        >
          <svg
            className="w-full cursor-pointer mr-2 "
            fill="none"
            height="36"
            viewBox="0 0 59 36"
            width="59"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9.5"
              cy="18.5"
              r="5"
              stroke="currentColor"
              strokeWidth="2"
            />
            <path
              d="M13.5 18.6112H18.2324C18.6625 18.6112 19.0444 18.3362 19.1808 17.9283L20.7981 13.0919C21.1151 12.1442 22.4728 12.1962 22.7164 13.1654L27.0802 30.5328C27.3478 31.5979 28.8876 31.5157 29.0403 30.4282L32.3788 6.64882C32.5367 5.52407 34.1498 5.49149 34.353 6.60894L38.013 26.7361C38.191 27.7149 39.5338 27.8616 39.9189 26.9442L43.1601 19.2241C43.316 18.8528 43.6794 18.6112 44.0821 18.6112H48.5"
              stroke="currentColor"
              strokeWidth="5"
            />
            <circle
              cx="52.5"
              cy="18.5"
              r="5"
              stroke="currentColor"
              strokeWidth="2"
            />
          </svg>
          <h1
            className="text-3xl font-bold"
            data-testid="pulse"
          >
            PULSE
          </h1>
        </div>
        <div
          className="flex border-none justify-around items-center ml-16 mt-4 cursor-pointer"
        >
          <svg
            className=" fa-xs "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
            />
          </svg>
          <svg
            className=" "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
            />
          </svg>
          <svg
            className="mr-1 "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
          <svg
            className="mr-1 "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
            />
          </svg>
        </div>
      </div>
      <div
        className="flex flex-col py-4 lg:py-0 lg:flex-row"
      >
        <div
          className="flex py-4"
        >
          <ul
            className="lg:flex lg:flex-col mr-[8vh] md:mr-[16vh] lg:mr-0 lg:ml-28 cursor-pointer"
          >
            <li
              className="py-2 text-sm font-bold"
            >
              Dev Pulse
            </li>
            <li
              className="py-2 text-xs"
            >
              About us
            </li>
            <li
              className="py-2 text-xs"
            >
              Contact us
            </li>
          </ul>
          <ul
            className="lg:flex lg:flex-col ml-[8vh] md:ml-[16vh] lg:ml-28 cursor-pointer"
          >
            <li
              className="py-2 text-sm font-bold"
            >
              Product
            </li>
            <li
              className="py-2 text-xs"
            >
              Features
            </li>
            <li
              className="py-2 text-xs"
            >
              Integrations
            </li>
          </ul>
        </div>
        <div
          className="flex py-4"
        >
          <ul
            className="lg:flex lg:flex-col mr-[8vh] md:mr-[16vh] lg:mr-0 lg:ml-28 cursor-pointer"
          >
            <li
              className="py-2 text-sm font-bold"
            >
              Resources
            </li>
            <li
              className="py-2 text-xs"
            >
              Community
            </li>
            <li
              className="py-2 text-xs"
            >
              Help Center
            </li>
          </ul>
          <ul
            className="lg:flex lg:flex-col ml-[8vh] md:ml-[16vh] lg:ml-28 cursor-pointer"
          >
            <li
              className="py-2 text-sm font-bold"
            >
              Basic Settings
            </li>
            <li
              className="py-2 text-xs"
            >
              Terms and conditions
            </li>
            <li
              className="py-2 text-xs"
            >
              Privacy and Policies
            </li>
            <select
              className="mt-2 outline rounded-md px-2 py-1 dark:text-dark-text-fill dark:bg-dark-bg "
              data-testid="lanChange"
              defaultValue="en-US"
              name="language"
              onChange={[Function]}
            >
              <option
                value="en"
              >
                English
              </option>
              <option
                value="kn"
              >
                Kinyarwanda
              </option>
              <option
                value="fr"
              >
                Français
              </option>
            </select>
          </ul>
        </div>
      </div>
    </div>
    <div
      className=" lg:flex mr-10"
    >
      <span
        className="px-4 lg:py-3 cursor-pointer text-lg"
      >
        © 
        2024
         Pulse Technologies
      </span>
    </div>
  </div>
</div>
`;
